test/puppetlabs/trapperkeeper/services/status/status_service_test.clj

```diff
@@ -11,57 +11,83 @@
 
 (use-fixtures :once schema-test/validate-schemas)
 
+(def status-service-config
+  {:webserver {:port 8180
+               :host "0.0.0.0"}
+   :web-router-service
+     {:puppetlabs.trapperkeeper.services.status.status-service/status-service "/status"}})
+
+(defmacro with-status-service
+  "Macro to start the status service and its dependencies (jetty9 and
+  webrouting service), along with any other services desired."
+  [app services & body]
+  `(with-app-with-config
+    ~app
+    (concat [jetty9-service/jetty9-service
+             webrouting-service/webrouting-service
+             status-service] ~services)
+    status-service-config
+    (do ~@body)))
+
 (defservice foo-service
   [[:StatusService register-status]]
   (init [this context]
-    (register-status "foo" "1.1.0" 1 (fn [] "foo status 1"))
-    (register-status "foo" "1.1.0" 2 (fn [] "foo status 2"))
+    (register-status "foo" "1.1.0" 1 (fn [level] (str "foo status 1 " level)))
+    (register-status "foo" "1.1.0" 2 (fn [level] (str "foo status 2 " level)))
     context))
 
 (defservice bar-service
   [[:StatusService register-status]]
   (init [this context]
-    (register-status "bar" "0.1.0" 1 (fn [] "bar status 1"))
+    (register-status "bar" "0.1.0" 1 (fn [level] (str "bar status 1 " level)))
     context))
 
 (deftest rollup-status-endpoint-test
-  (with-app-with-config
+  (with-status-service
     app
-    [jetty9-service/jetty9-service
-     webrouting-service/webrouting-service
-     status-service
-     foo-service
+    [foo-service
      bar-service]
-    {:webserver {:port 8180
-                 :host "0.0.0.0"}
-     :web-router-service {:puppetlabs.trapperkeeper.services.status.status-service/status-service "/status"}}
-    (let [req (http-client/get "http://localhost:8180/status/v1/services")
-          body (json/parse-string (slurp (:body req)))]
-      (is (= 200 (:status req)))
-      (is (= {"bar" {"service-version" "0.1.0"
-                     "service-status-version" 1
-                     "status" "bar status 1"}
-              "foo" {"service-version" "1.1.0"
-                     "service-status-version" 2
-                     "status" "foo status 2"}}
-             body)))))
+    (testing "returns latest status for all services"
+      (let [req (http-client/get "http://localhost:8180/status/v1/services")
```

Comment:
Maybe rename this var to `resp`?  I was confused momentarily because I thought you were reading the body of the request rather than the response.